### PR TITLE
Enable retries by default

### LINKED ISSUE #520: Add documentation for retry configuration in pystac-client and enable them by default
Please refer the [API error](https://github.com/microsoft/planetary-computer-apis/issues/176) issue for the problems faced.

### PR DESCRIPTION
**Pull request by gadomski**
**Related Issue(s):** 

- Closes #520 

**Description:**

Turned out to be simpler than I expected. I also added some docs around more complication configuration.

I moved some of the doc sections around to a way that makes more sense than me.

**PR Checklist:**

- [x] Code is formatted
- [x] Tests pass
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac-api-client/blob/main/CHANGELOG.md)

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -117,6 +117,72 @@ there are no ``"conformsTo"`` uris set at all. But they can be explicitly set:
...
+   ...    modifier=planetary_computer.sign_inplace,
+   ... )
+   >>> item = next(catalog.get_collection("sentinel-2-l2a").get_all_items())
+   >>> requests.head(item.assets["B02"].href).status_code
+   200
+
+Without the modifier, we would have received a 404 error because the asset
+is in a private storage container.
+
+``pystac-client`` expects that the ``modifier`` callable modifies the result
```
Comment (jsignell) [inline docs/usage.rst:166]: Would it make sense to sketch out a custom modifier?

**Review by jsignell — APPROVED** [review]
This looks good, just one suggestion

File: docs/usage.rst (hunk 1 above)
Comment (gadomski) [reply docs/usage.rst:166]: Makes sense, while we're touching these lines. I'll add.

Comment (gadomski) [reply docs/usage.rst:166]: Doc update blocked by https://github.com/stac-utils/pystac-client/issues/533

**Pull request merged**